### PR TITLE
[WIP] Add exokit-home keyboard+menu for MLO

### PR DESCRIPTION
**Pull request by chrislatorres**
This PR ports the keyboard + url menu from exokit-home to be used with Magic Leap One.

- [x] add controller rays/pointers
- [x] add controller position tracking
- [x] add keyhandling with siteUrl
- [ ] add ray/pointer key interaction under `onselect`
- [ ] debug canvas-text related things that are causing crash:
```
ctx.font = `${fontSize}px Arial`;
ctx.fillText(urlState.text, 10, urlBarHeight - 10*2 - 30);
const {width: barWidth} = ctx.measureText('[');
const {width} = ctx.measureText('[' + urlState.text.slice(0, i) + ']');
```

-------
dp70skoucaawm_p

### CODE REVIEW

File: examples/keyboardurl_ml.html [hunk 1]
```diff
@@ -0,0 +1,636 @@
...
+      session.requestAnimationFrame((timestamp, frame) => {
+        renderer.vr.setSession(session, {
+          frameOfReferenceType: 'stage',
+        });
+
+          let select = false;
+          session.onselect = () => {
+            console.log('onselect');
+            selected = true;
+            // controllerMesh.rayMesh.material.color.setHex(selected ? RAY_HIGHLIGHT_COLOR : RAY_COLOR);
```
Comment (modulesio) [inline examples/keyboardurl_ml.html]: This is probably a good place to emit the "current key".

**Comment by chrislatorres** [comment]
The backtrace from gdb debugging the font crash: 
screenshot from 2018-10-20 16-49-05

**Comment by modulesio** [comment]
Thanks, I figured it was the fonts. I actually don't know what the native font support is on MLO, but Skia seems to be exploding on this.

Either we can use the Android font system or we need to use some other supported libs.

Next steps is probably:
- Figure out the exact call the explodes, with skia logging
- Research why/what alternatives there are
- Move to a font system that _is_ in the collection of libs in the MLSDK, which I'm presuming there is

**Comment by modulesio** [comment]
Inspirational code from Servo: https://github.com/servo/servo/pull/21985/files#diff-d526bc10d7ec14d71e70023ce474ea3fR130

**Comment by chrislatorres** [comment]
Freetype work is being PR'd at: https://github.com/modulesio/skia/pull/2

**Comment by chrislatorres** [comment]
Closing now, the work was completed at #547

**Comment by chrislatorres** [comment]
Closing now, the work was completed at #547

**Comment by chrislatorres** [comment]
Closing now, the work was completed at #547

**Comment by chrislatorres** [comment]
Closing now, the work was completed at #547

**Comment by chrislatorres** [comment]
Closing now, the work was completed at #547

**Comment by chrislatorres** [comment]
Closing now, the work was completed at #547

**Comment by chrislatorres** [comment]
Closing now, the work was completed at #547

**Pull request closed without merging**